pkg/chrootarchive/archive.go

```diff
@@ -48,16 +46,20 @@ func untarHandler(tarArchive io.Reader, dest string, options *archive.TarOptions
 		}
 	}
 
-	r := ioutil.NopCloser(tarArchive)
+	r := tarArchive
```

Comment:
Yes, but the comments are still there in [previous commit](https://github.com/c0b/docker/commit/6aa28ae99cc94771bf09dd79d2d4c99c49fcc6c1#diff-45e15b27a574e8c07552aad3688b3142) but I removed comments and the defer statement; do you prefer to keep it in source code?

While, squash everything together doesn't look better; this whole changeset actually fixed two different race described in issue #16924 comment-1 and comment-2; My initial commits kept fixing it separately and look better
1) Change CmdStream to assign cmd.Stdin Stdout Stderr instead of do io.Copy in goroutines, because that's exactly the "os/exec" does; this commit only does simplifying;
2) fixed race-1 in 4223cd9 that's the race caused daemon crash, it's on bufio between defer decompressedArchive.Close() which calls pools.Put(buf) and CmdStream goroutine which is still doing io.Copy
the fix is done by returning a Done() receive only channel (inspired by golang.org/x/net/context) and let the Close code wait on that;
3) fix race-2 in io.File, between pkg/archive.(*Archiver).UntarPath() which has a defer'ed call to archive.Close(), while exec.Cmd has a goroutine doing io.Copy at the same time ;
4) later on I realized that Done() interface is not necessary, removed it and just implement Read / Close on the wrapper type.

I can fixup 4 merge into 2, and make 3 commits will be clear;